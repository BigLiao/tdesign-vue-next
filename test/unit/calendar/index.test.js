import { mount } from '@vue/test-utils';
import Calendar from '@/src/calendar/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Calendar', () => {
  // test props api
  describe(':props', () => {
    it('', () => {
      const wrapper = mount({
        render() {
          return <Calendar></Calendar>;
        },
      });
      expect(wrapper.isEmpty()).toBe(false);
    });

    it(':value', () => {
      const wrapper = mount({
        render() {
          return <Calendar value={'1998-11-11'}></Calendar>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':mode', () => {
      const wrapper = mount({
        render() {
          return <Calendar mode={'year'}></Calendar>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':firstDayOfWeek', () => {
      const wrapper = mount({
        render() {
          return <Calendar firstDayOfWeek={3}></Calendar>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':isShowWeekendDefault', () => {
      const wrapper = mount({
        render() {
          return <Calendar isShowWeekendDefault={false}></Calendar>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':isShowWeekendDefault', () => {
      const wrapper = mount({
        render() {
          const testRange = {
            from: '2018-08',  // new Date(2017, 7)
            to: '2028-04',  // new Date(2027, 3)
          };
          return <Calendar range={testRange}></Calendar>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':theme', () => {
      const wrapper = mount({
        render() {
          return <Calendar theme={'card'}></Calendar>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test events
  describe('@event', () => {
    window.console.info('calendar test @event');
  });

  // test slots
  describe('<slot>', () => {
    it('', () => {
      window.console.info('calendar test<slot>');
    });
  });

  // test exposure function
  describe('function', () => {
    it('', () => {
      window.console.info('calendar test function');
    });
  });
});
