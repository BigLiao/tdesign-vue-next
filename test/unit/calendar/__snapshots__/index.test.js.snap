// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar :props :firstDayOfWeek 1`] = ``;

exports[`Calendar :props :isShowWeekendDefault 1`] = ``;

exports[`Calendar :props :mode 1`] = ``;

exports[`Calendar :props :range 1`] = ``;

exports[`Calendar :props :theme 1`] = ``;

exports[`Calendar :props :value 1`] = ``;
